src/vivarium_cluster_tools/vipin/log_parser.py

```diff
@@ -0,0 +1,76 @@
+from loguru import logger
+from pathlib import Path
+from random import sample
+import pandas as pd
+import json
+import re
+from pandas.io.json import json_normalize
+from typing import Union
+
+
+class WorkerLog:
+
+    def __init__(self, file: Path):
+        self.file = file
+        name = file.stem.split('.')  # TODO: validate name is <sge job id>.<task id>.log
+        self.sge_job_id = int(name[0])
+        self.task_id = int(name[1])
+
+    def get_summaries(self) -> dict:
+        """Get all performance summary log messages in WorkerLog"""
+        telemetry_pattern = re.compile(r'^\{\'host\'.+\'job_id\'.+\}$')
+
+        # Ideally we'd only need to look at the tail of the file, but because a worker can do multiple draws, we need
+        # to iterate over all the lines...
+        # TODO: either change worker to cut a new log for each draw and check tail reversed(list(log))
+        #  *OR* parallelize this
+        f = self.file.open('r')
+        for line in f.readlines():
+            message = json.loads(line)['record']['message']
+            m = telemetry_pattern.fullmatch(str(message))
+            if m:
+                yield json_normalize(message, sep='_')
+        f.close()
+
+
+def parse_log_directory(input_directory: Union[Path, str], output_directory: Union[Path, str], output_hdf: bool,
+                        random_draw: int = 0):
+    input_directory, output_directory = Path(input_directory), Path(output_directory)
+    worker_log_pattern = re.compile(r'^([0-9]+)\.([0-9]+)\.log$')
+    log_files = [f for f in input_directory.iterdir() if worker_log_pattern.fullmatch(f.name)]
+
+    if random_draw:
+        log_files = sample(log_files, random_draw)
+
+    worker_data = []
+    for i, file in enumerate(log_files):
```

Comment:
Would it make sense to report every 10th, 20th, etc. file as opposed to every file?